### PR TITLE
Fix file uploader, test typed attributes

### PR DESCRIPTION
**Pull request by bizley**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️ see below
| Fixed issues  | #19855 

Validator does not modify the attribute anymore - it was filtering out non-uploaded-files values from the array so far, but only for the array typed attributes with maxFiles != 1 or minFiles > 1.
Setting max files != 1 and sending empty array does not ends in error.

### CODE REVIEW

File: tests/framework/validators/FileValidatorTest.php [hunk 1]
```diff
@@ -314,6 +315,32 @@ public function testValidateAttribute_minFilesTwoMaxFilesUnlimited_hasError()
...
+                    'name' => 'image.png',
+                    'size' => 1024,
+                    'type' => 'image/png',
+                ],
+            ]
+        )[0];
+        $model = FakedValidationModel::createWithAttributes(['attr_images' => [$files]]);
+
+        $validator->validateAttribute($model, 'attr_images');
+        $this->assertFalse($model->hasErrors('attr_images'));
```
[the remark this answers is not in the document]
Comment (bizley) [reply tests/framework/validators/FileValidatorTest.php:341]: No, there is `[0]` on line 337, only one file.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Would you please add a line for CHANGELOG?

**Review by Arhell — APPROVED** [review]

**Comment by samdark** [comment]
:+1:

File: framework/validators/FileValidator.php [hunk 2]
```diff
@@ -208,40 +208,23 @@ public function init()
...
-            foreach ($files as $file) {
-                $result = $this->validateValue($file);
-                if (!empty($result)) {
-                    $this->addError($model, $attribute, $result[0], $result[1]);
-                }
-            }
-        } else {
-            $result = $this->validateValue($model->$attribute);
+        foreach ($files as $file) {
+            $result = $this->validateValue($file);
```
Comment (marcovtwout) [inline framework/validators/FileValidator.php:227]: @bizley validateValue() is now not called anymore when submitting an empty form, I created an issue with further details: https://github.com/yiisoft/yii2/issues/20231

**Pull request merged**